src/schema/v2/unleashFeatureFlag.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  GraphQLObjectType,
+  GraphQLNonNull,
+  GraphQLString,
+  GraphQLList,
+  GraphQLInt,
+} from "graphql"
+import { ResolverContext } from "types/graphql"
+
+/**
+ * A client-facing representation of an Unleash feature flag.
+ */
+export const UnleashFeatureFlag = new GraphQLObjectType<any, ResolverContext>({
```

Comment:
Ah @anandaroop - we already have all of this stuff wired up for tools.artsy.net: 

https://github.com/artsy/metaphysics/blob/main/src/schema/v2/admin/featureFlags.ts

Can we update things there?

Reply:
Also tho I don't see us abandoning feature flags just yet (though I have been exploring posthog) ideally we don't encode service names in variables and keep things generic.

Reply:
Also, wanted to say, if `admin` isn't the right parent folder, feel free to move it out to just like `featureFlags`! But `admin` does feel kinda right for this kind of stuff (or something similar)

Reply:
Also, wanted to say, if `admin` isn't the right parent folder, feel free to move it out to just like `featureFlags`! But `admin` does feel kinda right for this kind of stuff (or something similar)

Reply:
Not actually the same type, see [above](https://github.com/artsy/metaphysics/pull/6160#discussion_r1814183706)!

- MP's Unleash-related admin fields use data loaders to hit the _admin_ REST API directly
  - e.g. `https://unleash.artsy.net/api/admin/projects/default/features` 
  - that^ returns data for administering flags

- MP's `lib/featureFlags` uses the Unleash API client to hit the _client_ REST API
  - e.g. `https://unleash.artsy.net/api/client/features/`
  - that^ returns data for clients which need only to evaluate flags

So: different endpoints with different [access](https://github.com/artsy/metaphysics/blob/4c8a2e1c556eeb251bbc575c65c55535c9392ecf/src/config.ts#L79) [credentials](https://github.com/artsy/metaphysics/blob/4c8a2e1c556eeb251bbc575c65c55535c9392ecf/src/config.ts#L82) and different response shapes, see e.g. [here](https://docs.getunleash.io/reference/api/unleash/get-features) vs [here](https://docs.getunleash.io/reference/api/unleash/get-all-client-features).

It _is_ confusing that these appear so similar, but they represent different views into Unleash.

Reply:
So the question is: can this be consolidated? Whats the difference?

Reply:
Here's what I'm going to do:

- consolidate the admin- and client-facing representations of the Unleash flags into a shared location:
  - `schema/v2/featureFlags/admin` 
  - `schema/v2/featureFlags/client`

- rename the new client facing type from `UnleashFeatureFlag` to `ClientFeatureFlag`

I didn't see specs for the admin stuff, but I manually verified via local Forque.

I still see these as separate types since they map to separate API resources, and I don't see much benefit of trying to merge them. (But I don't object if someone wants to try and merge these. That should be a little easier since everything is co-located now).

Gonna roll with this for now so we can move on to the Eigen side of this.